pkg/utils/gardener/seed.go

```diff
@@ -103,6 +103,34 @@ func ComputeRequiredExtensionsForSeed(seed *gardencorev1beta1.Seed, controllerRe
 	return wantedKindTypeCombinations
 }
 
+// ComputeEnabledTypesForKindExtensionSeed computes the enabled extension types for a given Seed and ControllerRegistrationList.
+// It considers extensions explicitly enabled or disabled in the Seed specification and those automatically enabled
+// based on the ControllerRegistration resources.
+func ComputeEnabledTypesForKindExtensionSeed(seed *gardencorev1beta1.Seed, controllerRegistrationList *gardencorev1beta1.ControllerRegistrationList) sets.Set[string] {
```

Comment:
This looks duplicative to `ComputeEnabledTypesForKindExtension`. The only difference is the condition under which extensions are globally enabled.
Should we deduplicate this?